pages/empress/js/callbacks.js

```diff
@@ -720,3 +764,115 @@ function downloadLinks(clsID) {
 function downloadSubtree(clsID) {
     downloadText(tree.toNewick(clsID) + ";", clsID + ".nwk");
 }
+
+
+/**
+ * Perform quick search.
+ * @function quickSearch
+ */
+function quickSearch() {
+  let bar = document.getElementById("quick-search");
+  let keyword = bar.value;
+  bar.value = "";
+
+  // obtain and sort candidate Ids
+  let taxPrefix = getTaxPrefix();
+  let metaKeys = Object.keys(tree.metadata);
+  let ids = metaKeys.filter(function(x) {
+    return x.startsWith("G");
+  });
+  ids.sort(sort2ndNum);
+  if (taxPrefix.endsWith("c_")) {
+    let nIds = metaKeys.filter(function(x) {
+      return x.startsWith("N") && x !== "N1";
+    });
+    nIds.sort(sort2ndNum);
+    ids = nIds.concat(ids);
+  }
+
+  // search for keyword
+  let target = searchKeyword(keyword, taxPrefix, ids);
+
+  // not found
+  if (!target) {
+    toastMsg("Keyword \"" + keyword + "\" is not found.");
+    return;
+  }
+
+  // found but invisible
+  if (!tree.metadata[target]["branch_is_visible"]) {
```

Comment:
@kwcantrell Here is how I attempt to check if invisible. It works when the node is in a triangle. But it won't help if the node is out of the viewport. Not urgent.